src/python/pants/jvm/dependency_inference/symbol_mapper.py

```diff
@@ -26,41 +28,56 @@ class JvmFirstPartyPackageMappingException(Exception):
     pass
 
 
+class SymbolNamespace(Enum):
+    """Represents a "namespace" for a symbol, i.e. is the symbol a generic JVM name or did it
+    originate from a particular language (and may be encoded for JVM purposes)."""
+
+    JVM = "jvm"
+    SCALA = "scala"
```

Comment:
This seems fine... but it seems like it could also become multiple different mappings instead, which might reduce complexity of this class (which could maybe become a `trie` at some point). That's the approach I took in https://github.com/pantsbuild/pants/issues/13621#issuecomment-974263816

More generally though, I don't think it will ever be possible to classify thirdparty deps (except maybe by parsing and de-mangling their symbols), so I wonder whether it is actually worth it to try and namespace them like this. I'm also not seeing the pitfall to Scala pulling in a mangled symbol.

Reply:
Okay will just lump them together for now. We can always revisit an approach like this if it actually becomes necessary.